src/plugins/saved_objects_management/common/types/types.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type {
+  SavedObjectGetRelationshipsResponseV1,
+  SavedObjectInvalidRelationV1,
+  SavedObjectManagementTypeInfoV1,
+  SavedObjectRelationV1,
+  SavedObjectRelationKindV1,
+  SavedObjectMetadataV1,
+  SavedObjectWithMetadataV1,
+} from './v1';
```

Comment:
Keeping the version-specific types within a plugin means that plugin authors will need to implement their own safeguards against changes to "legacy" versions. Obviously we can't recommend a generic control mechanism and each team will need to implement their own restrictions.

Reply:
>  linting to prevent future modifications to older/previous versions

Yeah, this would be a really nice!

>  a /* */ comment above each type

Sounds fairly pragmatic :) .

I think we could get quite far with convention here too. If you have a `README.md` or a specific directory structure most developers should be able to pick the pattern: if you want to change something here, you must create a new version of the interface.

Then, in the future, a control like what @TinaHeiligers described would be 🥇